test/pylib/pool.py

```diff
@@ -13,46 +13,72 @@ class Pool(Generic[T]):
     builder async function to build a new object.
 
     Usage example:
-    async def start_server():
-        return Server()
-    pool = Pool(4, start_server)
-    ...
-    async with pool.instance() as server:
-        await run_test(test, server)
-    """
+        async def start_server():
+            return Server()
+        pool = Pool(4, start_server)
+
+        server = await pool.get()
+        try:
+            await run_test(test, server)
+        finally:
+            await pool.put(server)
 
-    def __init__(self, size: int, build: Callable[[], Awaitable[T]]):
-        assert(size >= 0)
-        self.pool: asyncio.Queue[T] = asyncio.Queue(size)
-        self.build = build
-        self.total = 0
+    Alternatively:
+        async with pool.instance() as server:
+            await run_test(test, server)
+
+    If the object is considered no longer usable by other users of the pool
+    you can 'steal' it, which frees up space in the pool.
+        server = await.pool.get()
+        dirty = True
+        try:
+            dirty = await run_test(test, server)
+        finally:
+            if dirty:
+                await pool.steal()
+            else:
+                await pool.put(server)
```

Comment:
I think the analogies of "getting" and "stealing" from a "pool" isn't clear.
What I missed in first reading (and only understood in second reading), is the the pool deliberately has a fixed maximum size, so if N users "borrowed" (get()) objects the next one will block until one of the existing borrowers will either return the object, or admit it was stolen (in which case a new one can be created).
But I guess we can improve the documentation later.